ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/Template2.kt

```diff
@@ -115,10 +127,43 @@ private fun SelectPackageButton(
         shape = RoundedCornerShape(15.dp),
         border = border,
     ) {
-        Text(
-            text = "Purchase ${packageInfo.rcPackage.identifier}. " +
-                "Price: ${packageInfo.rcPackage.product.price.formatted}",
-        )
+        Column(
+            modifier = Modifier.fillMaxSize().padding(4.dp),
+            verticalArrangement = Arrangement.spacedBy(4.dp),
+        ) {
+            Row(horizontalArrangement = Arrangement.spacedBy(6.dp), verticalAlignment = Alignment.CenterVertically) {
+                CheckmarkBox(isSelected = isSelected, colors = state.currentColors)
+                Text(
+                    text = packageInfo.localization.offerName ?: packageInfo.rcPackage.product.title,
+                )
+            }
+            // TODO-PAYWALLS: Substitute with intro eligibility composable
+            val pricingPhases = packageInfo.rcPackage.product.defaultOption?.pricingPhases
+            val offerText = if (pricingPhases != null && pricingPhases.size > 1) {
+                packageInfo.localization.offerDetailsWithIntroOffer
+            } else {
+                packageInfo.localization.offerDetails
+            }
```

Comment:
Let's extract this logic to the composable in #1286? :)

Reply:
yup, I left a comment indicating that. Now that it's merged, I can try to move it

Reply:
Done in https://github.com/RevenueCat/purchases-android/pull/1301/commits/2d9d1e1b3f634048c2d132b3d2903648503c86e8